dali/ft/filecopy.cpp

```diff
@@ -830,8 +830,12 @@ void FileSprayer::assignPartitionFilenames()
             cur.inputName.set(sources.item(cur.whichInput).filename);
             setCanAccessDirectly(cur.inputName);
         }
-        cur.outputName.set(targets.item(cur.whichOutput).filename);
-        setCanAccessDirectly(cur.outputName);
+        // Prevent to use invalid (-1) whichOutput index.
+        if (cur.whichOutput != (unsigned)-1)
```

Comment:
I am not entirely sure at the moment, how a PartitionPoint with a whichOutput of -1 is getting added at the moment,
hence I'm not sure if this change should be necessary. See my other comment in CPartitioner::commonCalcPartitions() in daftformt.cpp

Reply:
Changed to throw and error.